src/test/regress/expected/single_shard_table_udfs.out

```diff
@@ -557,5 +572,112 @@ WHERE
 (3 rows)
 
 END;
+-- test citus_shard_cost_by_disk_size
+CREATE TABLE size_tbl_dist (a INT, b TEXT);
+SELECT create_distributed_table('size_tbl_dist', 'a', shard_count:=4, colocate_with:='none');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+CREATE TABLE size_tbl_single (a INT, b TEXT);
+SELECT create_distributed_table('size_tbl_single', NULL, colocate_with:='none');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+INSERT INTO size_tbl_dist SELECT 1, '1234567890' FROM generate_series(1, 10000);
+INSERT INTO size_tbl_single SELECT 1, '1234567890' FROM generate_series(1, 10000);
+SELECT citus_shard_cost_by_disk_size(c1.shardid) = citus_shard_cost_by_disk_size(c2.shardid) AS equal_cost
+FROM citus_shards c1, citus_shards c2
+WHERE c1.table_name::TEXT = 'size_tbl_dist' AND c2.table_name::TEXT = 'size_tbl_single'
+ORDER BY c1.shard_size DESC
+LIMIT 1;
+ equal_cost
+---------------------------------------------------------------------
+ t
+(1 row)
+
+-- test update statistics UDFs
+TRUNCATE size_tbl_dist, size_tbl_single;
+SELECT shardid AS shard_single
+FROM citus_shards
+WHERE table_name::TEXT = 'size_tbl_single'
+LIMIT 1 \gset
+SELECT get_shard_id_for_distribution_column('size_tbl_dist', 1) AS shard_dist  \gset
+SELECT dsp1.shardlength, dsp2.shardlength
+FROM pg_dist_shard_placement dsp1, pg_dist_shard_placement dsp2
+WHERE dsp1.shardid = :shard_single AND dsp2.shardid = :shard_dist
+LIMIT 1;
+ shardlength | shardlength
+---------------------------------------------------------------------
+           0 |           0
+(1 row)
+
+INSERT INTO size_tbl_dist SELECT 1, '1234567890' FROM generate_series(1, 10000);
+INSERT INTO size_tbl_single SELECT 1, '1234567890' FROM generate_series(1, 10000);
+SELECT dsp1.shardlength, dsp2.shardlength
+FROM pg_dist_shard_placement dsp1, pg_dist_shard_placement dsp2
+WHERE dsp1.shardid = :shard_single AND dsp2.shardid = :shard_dist
+LIMIT 1;
+ shardlength | shardlength
+---------------------------------------------------------------------
+           0 |           0
+(1 row)
+
+SELECT citus_update_table_statistics('size_tbl_dist');
+ citus_update_table_statistics
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT citus_update_table_statistics('size_tbl_single');
+ citus_update_table_statistics
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT dsp1.shardlength, dsp2.shardlength
+FROM pg_dist_shard_placement dsp1, pg_dist_shard_placement dsp2
+WHERE dsp1.shardid = :shard_single AND dsp2.shardid = :shard_dist
+LIMIT 1;
+ shardlength | shardlength
+---------------------------------------------------------------------
+      483328 |      483328
```

Comment:
sizes sometimes vary (e.g. across PG versions, or after autovacuum), can we just do > 0?

Reply:
Sure, then I'm also removing the dist table's checks.